syft/frameworks/torch/tensors/interpreters/autograd.py

```diff
@@ -45,29 +43,33 @@ def __init__(
 
     def backward(self, grad=None):
         if grad is None:
-            grad = torch.ones(self.child.shape)
-
+            # Build a torch tensor of ones with the same shape
+            # And chain structure than self
+            grad = self * 0 + 1
         backwards_grad(self.grad_fn, grad)
 
+    @property
+    def data(self):
+        return self
+
+    @data.setter
+    def data(self, new_data):
+        self.child = new_data.child
+        return self
+
     @property
     def grad(self):
         return self._grad
 
     @grad.setter
     def grad(self, value):
-        if value is not None:
-            self.child.setattr("grad", value)
-            self._grad = value.wrap()
-        else:
-            self._grad = value
-
-        if self.parent is not None:
-            # self.parent is a weakref
-            self.parent().grad = self._grad
+        self._grad = value
 
     def attr(self, attr_name):
-        attr_val = self.child.attr(attr_name)
+        if attr_name == "grad":
+            return self.grad
```

Comment:
I am not sure I understand everything so I'll just ask this question here:
Why is grad accessible from the other types of tensors (FPT, AST, native, etc.)? Shouldn't having it in the AutogradTensor wrapping everything be sufficient?

Reply:
So it depends, for native tensor or any syft tensor which could be above an autograd tensor, we need to keep .grad to propagte the information,
For FPT and AST they can't be above the autograd node in the syft chain, so I specified
```
@property
    def grad(self):
        return None
```
It makes it clear and I think is useful for the serialization. Maybe there is a better way to handle .grad however, I haven't tried to achieve the most compact form